ehrapy/tools/cohort_tracking/_cohort_tracker.py

```diff
@@ -0,0 +1,332 @@
+import copy
+from typing import Any, Union
+
+import graphviz
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import seaborn as sns
+from scanpy import AnnData
+from tableone import TableOne
+
+
+def _check_columns_exist(df, columns):
+    if not all(col in df.columns for col in columns):
+        missing_columns = [col for col in columns if col not in df.columns]
+        raise ValueError(f"Columns {missing_columns} not found in dataframe.")
+
+
+# from tableone: https://github.com/tompollard/tableone/blob/bfd6fbaa4ed3e9f59e1a75191c6296a2a80ccc64/tableone/tableone.py#L555
+def _detect_categorical_columns(data) -> list:
```

Comment:
I thought that already had a similar function somewhere in the `encoding` code. It'd be great if we could use a single implementation of this concept if my memory is right.

Reply:
Yeah, harmonizing both implementations to use the more robust and maintainable option would be great!

Reply:
Crosslinked to address this when taking care of issue #662.